File: 70-climbing-stairs/climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        a=1
        s=0
        b=2
        c=3
        if(n==0):
            return 0
        if(n==1):
            return 1
        if(n==2):
            return 2
        for i in range(3,n+1):
            s=a+b
            a=b
            b=s
        return s

File: 70-climbing-stairs/climbing-stairs.cpp
class Solution {
public:

    int f(int n,unordered_map<int,int>&tbl){
        if(n==0)
        return 1;
        if(n==1)
        return 1;
        if(tbl.find(n)==tbl.end()){
            int l = f(n-1,tbl);
            int r = f(n-2,tbl);
            tbl[n]=l+r;
        }
        return tbl[n];
    }

    int climbStairs(int n) {
        unordered_map<int,int>tbl;
        return f(n,tbl);
    }
};